src/unitquaternion.jl

```diff
@@ -57,11 +57,11 @@ function QuatRotation(w,x,y,z, normalize::Bool = true)
     QuatRotation{eltype(types)}(w,x,y,z, normalize)
 end
 
-function QuatRotation(q::T) where T<:Quaternion
-    if q.norm
-        return QuatRotation(real(q), imag_part(q)..., false)
+function QuatRotation(q::Quaternion{T}, normalize::Bool = true) where T<:Real
```

Comment:
It looks like this method should just forward to `QuatRotation{float(T)}(q,normalize)`, rather then replicate its functionality.